alica_engine/src/engine/allocationauthority/AllocationDifference.cpp

```diff
@@ -13,155 +13,154 @@ namespace alica
 {
 
 AllocationDifference::AllocationDifference()
+    :_reason(Reason::empty)
 {
-    this->reason = Reason::empty;
 }
 
 AllocationDifference::~AllocationDifference()
 {
-    // both are now shared pointers
-    //		for(auto i : this->additions)
-    //		{
-    //			delete i;
-    //		}
-    //		for(auto i : this->subtractions)
-    //		{
-    //			delete i;
-    //		}
 }
 
-AllocationDifference::Reason AllocationDifference::getReason()
+AllocationDifference::Reason AllocationDifference::getReason() const
 {
-    return reason;
+    return _reason;
 }
 
 void AllocationDifference::setReason(AllocationDifference::Reason reason)
 {
-    this->reason = reason;
+    _reason = reason;
 }
 
 /**
  * Returns whether the difference is empty, i.e., the corresponding allocations are the same
  * @return A bool
  */
-bool AllocationDifference::isEmpty()
+bool AllocationDifference::isEmpty() const
 {
-    return this->additions.size() == 0 && this->subtractions.size() == 0;
+    return _additions.empty() && _subtractions.empty();
 }
 
 /**
  * Reset this difference to the empty difference
  */
 void AllocationDifference::reset()
 {
-    // both are now shared pointers
-    //		for(auto i : this->additions)
-    //		{
-    //			delete i;
-    //		}
-    //		for(auto i : this->subtractions)
-    //		{
-    //			delete i;
-    //		}
-    this->additions.clear();
-    this->subtractions.clear();
-    this->reason = Reason::empty;
+    _additions.clear();
+    _subtractions.clear();
+    _reason = Reason::empty;
 }
 
 /**
  * Apply another difference to this one resulting in the composition of both
  * @param other A AllocationDifference*
```

Comment:
@param other A AllocationDifference&

Reply:
good catch!

Reply:
Actually, mentioning the type is redundant, it should rather be a description.
Also, the comment belongs into the header.